datus/tools/bi_tools/superset_adaptor.py

```
@@ -0,0 +1,306 @@
+        except httpx.HTTPError as exc:
+            raise SupersetAdaptorError(f"Superset API {method} {endpoint} failed: {exc}") from exc
+
+    def _ensure_authenticated(self) -> None:
+        if self._auth_header_value and self._token_expiration and time.time() < self._token_expiration:
+            return
+        self._authenticate()
+
+    def _authenticate(self) -> None:
+        if self.auth_type == AuthType.LOGIN:
+            if not isinstance(self.auth_params, dict):
+                raise SupersetAdaptorError("auth_params must be a mapping when using LOGIN auth_type")
```

Ruff (0.14.8)

Avoid specifying long messages outside the exception class

(TRY003)